forge/service_info.py

```diff
@@ -39,6 +39,7 @@
     Field("name", String(), default=OMIT, docs="The name to use for the container."),
     Field("context", String(), default=OMIT, docs="The build context."),
     Field("args", Map(String("string", "integer", "float")), default=OMIT, docs="Build arguments."),
+    Field("image_builder", String(), default=OMIT, docs="The docker image builder to be used: `docker` for `docker build`, `imagebuilder` for `openshift/imagebuilder`."),
```

Comment:
FYI, I believe you should be able to use a more constrained schema here, something like:

    Field("blah", Union(Constant("docker"), Constant("imagebuilder")), ...)

Also, this is a bit of a nit, but `image_builder: imagebulder` seems a bit odd. Would something like `type` or `bulder` make sense as the field name?

Reply:
@rhs Thanks for the suggestion!

A stricter schema sounds great. I'll look into it soon.

> Also, this is a bit of a nit, but image_builder: imagebulder seems a bit odd. Would something like type or bulder make sense as the field name?

At glance it did look odd to me, too :)

However, I ended up naming it so for consistency.

Given the parent configuration key is named `containers[]` while the class is named `CONTAINER`, I name it `image_builder` so that for example:

```yaml
containers:
  - image_builder: imagebuilder
```

can be read as `the first container uses the image_builder named "imagebuilder"`, as well as:

```python
container.image_builder
```

in python code would be read as `the container's image builder`.

If you named it `type`, `container.type = "imagebuilder"` would obviously be read as `the container's type is "imagebuilder"`. If you named it `builder`, `container.builder = "imagebuilder"` would be read as `the container's builder is "imagebuilder"`.

Both sounds not quite true for me. `docker build` and `imagebuilder` builds a docker image, not a docker container.

## Possible alternative

Would it look a bit nicer if it looked like this?

```yaml
containers:
- image:
     builder: imagebuilder
```